src/coreclr/vm/methodtable.cpp

```diff
@@ -2999,13 +3014,20 @@ bool MethodTable::IsLoongArch64OnlyOneField(MethodTable * pMT)
 
                 CorElementType fieldType = pFieldStart->GetFieldType();
 
-                bool isFixedBuffer = (CorTypeInfo::IsPrimitiveType_NoThrow(fieldType)
+                // InlineArray types and fixed buffer types have implied repeated fields.
```

Comment:
As above in the managed code, this code has a lot of duplicates. Find a way to refactor it.